### ISSUE TITLE
Should ScanExec use Spark-compatible cast instead of DataFusion cast?

### ISSUE THREAD

**Issue by andygrove** [apache/datafusion-comet#803]
### What is the problem the feature request solves?

In `ScanExec` there can be a difference between the Parquet data type and the Spark data type being requested. In this case, we currently delegate to DataFusion's `cast_with_options`, but should we be using Comet's `spark_cast` method instead?

**Comment by andygrove** [comment]
For complex type support, we likely need to fix this. We could maybe use the same SchemaAdapter approach that we are using in the Parquet exec proof-of-concept work.

**Comment by parthchandra** [comment]
IMHO `SchemaAdapter` is the right place to do this as the correction is applied before the vector is written.

**Comment by andygrove** [comment]
I agree that SchemaAdapter is now the solution. I will close this issue.
